### PR TITLE
Fix #762: Add a column for OTP attempt count

### LINKED ISSUE #762: Add a column for OTP attempt count
Currenly only failed attempt count is stored into the database. For reporting purposes it is useful to know all attempt count, so a separate column with verification attempt counter would be useful.

### PR DESCRIPTION
**Pull request by jandusil**
* Update table es_onboarding_otp -> new col 'total_attempts'
* Update 'total_attempts' count during various processes

### CODE REVIEW

File: docs/onboarding/Database-Structure.md [hunk 1]
```diff
@@ -69,21 +69,22 @@ Stores onboarding OTP codes used during activation and user verification.
...
-| `error_origin` | `VARCHAR(256)` |  | Origin of the error (`DOCUMENT_VERIFICATION`, `PRESENCE_CHECK`, `CLIENT_EVALUATION`, `OTP_VERIFICATION`, `PROCESS_LIMIT_CHECK`, `USER_REQUEST`). |
-| `failed_attempts` | `INTEGER` |  | Number of failed attempts for verification, |
-| `timestamp_created` | `TIMESTAMP` | `NOT NULL DEFAULT CURRENT_TIMESTAMP` | Timestamp when process was started. |
-| `timestamp_expiration` | `TIMESTAMP` | `NOT NULL` | Timestamp when the OTP expires. |
-| `timestamp_last_updated` | `TIMESTAMP` |  | Timestamp when record was last updated. |
-| `timestamp_verified` | `TIMESTAMP` |  | Timestamp when OTP was verified. |
-| `timestamp_failed` | `TIMESTAMP` |  | Timestamp when OTP failed. |
+| Name                     | Type | Info | Note                                                                                                                                             |
+|--------------------------|---|---|--------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (banterCZ) [inline docs/onboarding/Database-Structure.md:73]: If reformatted, I would expect everything. Or nothing at all.

File: docs/sql/oracle/onboarding/create-schema.sql [hunk 2]
```diff
@@ -51,6 +51,7 @@ CREATE TABLE ES_ONBOARDING_OTP (
     ERROR_DETAIL VARCHAR2(256 CHAR),
     ERROR_ORIGIN VARCHAR2(256 CHAR),
     FAILED_ATTEMPTS INTEGER,
+    TOTAL_ATTEMPTS INTEGER,
```
Comment (banterCZ) [inline docs/sql/oracle/onboarding/create-schema.sql:54]: The default value should be reflected here as well.

File: docs/sql/postgresql/onboarding/create-schema.sql [hunk 3]
```diff
@@ -55,6 +55,7 @@ CREATE TABLE es_onboarding_otp (
     error_detail VARCHAR(256),
     error_origin VARCHAR(256),
     failed_attempts INTEGER,
+    total_attempts INTEGER,
```
Comment (banterCZ) [inline docs/sql/postgresql/onboarding/create-schema.sql:58]: dtto

**Review by banterCZ — CHANGES_REQUESTED** [review]
A good job. Just details to improve.

File: docs/sql/oracle/onboarding/create-schema.sql [hunk 4]
```diff
@@ -51,6 +51,7 @@ CREATE TABLE ES_ONBOARDING_OTP (
     ERROR_DETAIL VARCHAR2(256 CHAR),
     ERROR_ORIGIN VARCHAR2(256 CHAR),
     FAILED_ATTEMPTS INTEGER,
+    TOTAL_ATTEMPTS INTEGER DEFAULT 0,
```
Comment (banterCZ) [inline docs/sql/oracle/onboarding/create-schema.sql:54]: Sorry, one more thing. It should also be mentioned in `PowerAuth-Enrollment-Onboarding-Server-1.5.0.md`. I know, it is a little duplicity with Liquibase, but the update process is not fine-tuned yet.

File: docs/onboarding/Database-Structure.md [hunk 5]
```diff
@@ -13,10 +13,10 @@ Onboarding Server uses ShedLock to synchronize scheduled operations. You need to
 
 ```sql
 CREATE TABLE shedlock (
-    name VARCHAR(64) NOT NULL PRIMARY KEY,
-    lock_until TIMESTAMP NOT NULL,
-    locked_at TIMESTAMP NOT NULL,
-    locked_by VARCHAR(255) NOT NULL
+                          name VARCHAR(64) NOT NULL PRIMARY KEY,
```
Comment (banterCZ) [inline docs/onboarding/Database-Structure.md:16]: Is this reformat needed?

File: docs/onboarding/Database-Structure.md [hunk 6]
```diff
@@ -71,19 +71,20 @@ Stores onboarding OTP codes used during activation and user verification.
...
-| `type` | `VARCHAR(32)` | `NOT NULL` | OTP code type (`ACTIVATION`, `USER_VERIFICATION`). |
-| `error_detail` | `VARCHAR(256)` |  | Detail of error (e.g. information about timeout or exceeded number of failed attempts). |
-| `error_origin` | `VARCHAR(256)` |  | Origin of the error (`DOCUMENT_VERIFICATION`, `PRESENCE_CHECK`, `CLIENT_EVALUATION`, `OTP_VERIFICATION`, `PROCESS_LIMIT_CHECK`, `USER_REQUEST`). |
-| `failed_attempts` | `INTEGER` |  | Number of failed attempts for verification, |
-| `timestamp_created` | `TIMESTAMP` | `NOT NULL DEFAULT CURRENT_TIMESTAMP` | Timestamp when process was started. |
-| `timestamp_expiration` | `TIMESTAMP` | `NOT NULL` | Timestamp when the OTP expires. |
-| `timestamp_last_updated` | `TIMESTAMP` |  | Timestamp when record was last updated. |
-| `timestamp_verified` | `TIMESTAMP` |  | Timestamp when OTP was verified. |
-| `timestamp_failed` | `TIMESTAMP` |  | Timestamp when OTP failed. |
+| `id`                     | `VARCHAR(36)` | `NOT NULL PRIMARY KEY` | Autogenerated record identifier (UUID).                                                                                                          |
```
Comment (banterCZ) [inline docs/onboarding/Database-Structure.md:74]: My idea was to reformat all columns of this single table or do not reformat it at all.

**Review by banterCZ — APPROVED** [review]
Good job.

**Review by romanstrobl — APPROVED** [review]
Looks good.

**Pull request merged**